services/frontend-service/src/ui/ReleaseDialog.tsx

```diff
@@ -129,99 +129,94 @@ export const randomLockId = () => 'ui-' + Math.random().toString(36).substring(7
 
 const LockButtonGroup = (props: {
     applicationName?: string;
-    openDialog?: () => void;
-    state?: string;
+    addToCart?: () => void;
+    inCart?: boolean;
     message: string;
     setMessage: (e: string) => void;
     open: boolean;
     setOpen: (e: boolean) => void;
 }) => {
-    const { applicationName, openDialog, state, message, setMessage, setOpen, open } = props;
+    const { applicationName, addToCart, inCart, message, setMessage, setOpen, open } = props;
     const updateMessage = React.useCallback((e) => setMessage(e.target.value), [setMessage]);
     const openInput = React.useCallback(() => setOpen(true), [setOpen]);
-    switch (state) {
-        case 'not-in-cart':
-            if (open) {
-                return (
-                    <Grow in={open} style={{ transformOrigin: 'right center' }}>
-                        {applicationName ? (
-                            <ButtonGroup className="overlay">
-                                <TextField label="Lock Message" variant="standard" onChange={updateMessage} />
-                                <IconButton onClick={openDialog} disabled={message === ''}>
-                                    <AddLockIcon />
-                                </IconButton>
-                            </ButtonGroup>
-                        ) : (
-                            <ButtonGroup className="overlay">
-                                <Button onClick={openDialog} disabled={message === ''}>
-                                    Add Lock
-                                </Button>
-                                <TextField label="Lock Message" variant="standard" onChange={updateMessage} />
-                            </ButtonGroup>
-                        )}
-                    </Grow>
-                );
-            } else {
-                return applicationName ? (
-                    <Tooltip title="Add lock">
-                        <IconButton onClick={openInput}>
-                            <AddLockIcon />
-                        </IconButton>
-                    </Tooltip>
-                ) : (
-                    <Button onClick={openInput}>Add Lock</Button>
-                );
-            }
-        case 'in-cart':
+    if (!inCart) {
```

Comment:
for readibility, I would move this around a bit:
```
if (inCart) {
   return ...
}
if (open) {
 ...
}
```
so that you have nested `if` statements